pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -87,6 +86,45 @@ func (service *AlertRuleService) CreateAlertRule(ctx context.Context, rule model
 	return rule, nil
 }
 
+// UpdateRuleGroup will update the interval for all rules in the group.
+func (service *AlertRuleService) UpdateRuleGroup(ctx context.Context, orgID int64, namespaceUID string, ruleGroup string, interval int64) error {
+	if err := service.validateRuleGroupInterval(interval); err != nil {
+		return err
+	}
+	return service.xact.InTransaction(ctx, func(ctx context.Context) error {
+		query := &models.ListAlertRulesQuery{
+			OrgID:         orgID,
+			NamespaceUIDs: []string{namespaceUID},
+			RuleGroup:     ruleGroup,
+		}
+		err := service.ruleStore.ListAlertRules(ctx, query)
+		if err != nil {
+			return err
+		}
+		updateRules := make([]store.UpdateRule, 0, len(query.Result))
+		for _, rule := range query.Result {
+			if rule.IntervalSeconds == interval {
+				continue
+			}
+			newRule := *rule
+			newRule.IntervalSeconds = interval
+			updateRules = append(updateRules, store.UpdateRule{
+				Existing: rule,
+				New:      newRule,
+			})
+		}
+		return service.ruleStore.UpdateAlertRules(ctx, updateRules)
+	})
+}
+
+func (service *AlertRuleService) validateRuleGroupInterval(interval int64) error {
+	if interval%service.defaultInterval != 0 || interval <= 0 {
+		return fmt.Errorf("%w: interval (%v) should be non-zero and divided exactly by scheduler interval: %v",
+			models.ErrAlertRuleFailedValidation, time.Duration(interval)*time.Second, service.defaultInterval)
+	}
+	return nil
+}
```

Comment:
This duplicates the logic in api. Please consider re-using that logic.

Reply:
Yeah, I got it from the store, extracted it now so both use the same.